microproxy/protocol/tls.py

```diff
@@ -70,17 +73,19 @@ def certificate_verify_cb(conn, x509, err_num, err_depth, verify_status):
 def create_dest_sslcontext(insecure=False, trusted_ca_certs="", alpn=None):
     ssl_ctx = create_basic_sslcontext()
 
-    if not insecure and trusted_ca_certs:
-        verify_mode = SSL.VERIFY_PEER | SSL.VERIFY_FAIL_IF_NO_PEER_CERT
+    if not insecure:
+        trusted_ca_certs = trusted_ca_certs or certifi.where()
         ssl_ctx.load_verify_locations(trusted_ca_certs)
+        ssl_ctx.set_verify(
+            SSL.VERIFY_PEER | SSL.VERIFY_FAIL_IF_NO_PEER_CERT,
+            certificate_verify_cb)
     else:
-        verify_mode = SSL.VERIFY_NONE
+        def verify_cb_always_true(*args, **kwargs):
```

Comment:
If I remember correctly there is no need to define a new function here. When the verify mode is ```SSL.VERIFY_NONE```, the exception will not raise no matter what the callback function returns when ```do_handshake```.